#!/bin/bash -l

#SBATCH -A g2020008
#SBATCH -p core
#SBATCH -n 4
#SBATCH -t 24:00:00
#SBATCH -J all-pac2ill
#SBATCH --mail-type=ALL
#SBATCH --mail-user meifang.wu.7427@student.uu.se

module load bioinfo-tools
module load FastQC
module load bwa
module load samtools
module load Pilon

dir=~/genome-analysis/analyses

# Although I won't need to quality check the Illumina sequence as they seem to already be trimmed. I will still run it through to see the quality post-trimmed.
fastqc $dir/data/raw_data/illumina_data/SRR6058604_scaffold_10.*P* --extract -o $dir/01*/illumina-fastqc/

# Align the paired illumina reads to a PacBio assembly generated by Canu. 
dir_align=$dir/01_preprocessing/illumina-bwa

bwa index $dir/02*/canu_assembly/durioz.contigs.fasta
bwa mem -t 8 \
    $dir/02*/canu_assembly/durioz.contigs.fasta $dir/data/raw_data/illumina_data/SRR6058604_scaffold_10.1P.fastq.gz $dir/data/raw_data/illumina_data/SRR6058604_scaffold_10.2P.fastq.gz > $dir_align/SRR6058604-aln.sam

# Analyze the alignments in SAM/BAM format with samtools
samtools view -b -@ 8 $dir_align/SRR6058604-aln.sam | \
    samtools sort > $dir_align/SRR6058604-aln-sorted.bam

rm $dir_align/SRR6058604-aln.sam

samtools index $dir_align/SRR6058604-aln-sorted.bam > $dir_align/SRR6058604-aln-sorted.bam.bai

# Pilon
pilon \
    --genome $dir/02*/canu_assembly/durioz.contigs.fasta \
    --frags $dir_align/SRR6058604-aln-sorted.bam \
    --changes \
    --output durioz.pilon \
    --outdir $dir/02_genome_assembly/pilon_assembly/ \
    --threads 8 \
    --diploid
